### ISSUE TITLE
VS Code Crashes from attempts to sign in

### ISSUE THREAD

**Issue by vregazzi** [microsoft/vscode#195800]
Does this issue occur when all extensions are disabled?: Yes/No
Yes

- VS Code Version: 1.83.1
- OS Version: Debian GNU/Linux 12 (bookworm) x86_64

Steps to Reproduce:

1. Open VS Code
2. Click `Backup and Settings Sync...` under gear
3. Click `Sign in` under new drop down menu
4. Application then errors with option to force quit

![image](https://github.com/microsoft/vscode/assets/118196285/16e54fa2-3f1d-431d-8235-9ee86bb426c9)

**Comment by rzhao271** [comment]
Thanks for the notice!
Could you create and attach a crash report following the instructions in https://github.com/microsoft/vscode/wiki/Native-Crash-Issues#creating-a-crash-report? The report will help us narrow down the issue.

**Comment by vregazzi** [comment]
No .dmp files are produced in the specified directory when I force quit. What else can I try to narrow this down?

**Comment by rzhao271** [comment]
I just noticed that it's a stall. I don't think the crash reporter will be able to grab anything in this case. @TylerLeonhardt are there any Linux dependencies required for auth that may have to be installed separately?

**Comment by TylerLeonhardt** [comment]
@vregazzi can you try launch VS Code with this verbose logging and share it with us:
```
code --verbose --vmodule="*/components/os_crypt/*=1"
```

**Comment by vregazzi** [comment]
[main 2023-10-17T23:44:45.114Z] menubarService#updateMenubar 1
[25344:1017/184445.695735:INFO:CONSOLE(645)] "%cTRACE color: #888 lifecycle: phase changed (value: 4)", source: vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js (645)
[25344:1017/184445.804512:INFO:CONSOLE(645)] "%cTRACE color: #888 RunAutomaticTasks: Trying to run tasks.", source: vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js (645)
[25344:1017/184445.824031:INFO:CONSOLE(645)] "%cTRACE color: #888 RunAutomaticTasks: Found 1 automatic tasks", source: vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js (645)
[main 2023-10-17T23:44:45.880Z] resolveShellEnv(): skipped (VSCODE_CLI is set)
[25344:1017/184446.166955:INFO:CONSOLE(645)] "%c INFO color: #33f [perf] Render performance baseline is 47ms", source: vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js (645)
[25379:1017/184447.919349:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
[25379:1017/184449.213256:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times!
[25379:1017/184449.220793:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 3 times!
[25344:1017/184453.572186:INFO:CONSOLE(645)] "%cTRACE color: #888 CommandService#executeCommand workbench.userDataSync.actions.turnOn", source: vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js (645)
[main 2023-10-17T23:44:55.624Z] resolveShellEnv(): skipped (VSCODE_CLI is set)
[25344:1017/184459.563790:INFO:CONSOLE(645)] "%cTRACE color: #888 onWillActivateByEvent:  onAuthenticationRequest:github", source: vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js (645)
[25344:1017/184459.605821:INFO:CONSOLE(645)] "%cTRACE color: #888 [mainThreadSecretState] Getting password for vscode.github-authentication extension:  github.auth", source: vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js (645)
[25344:1017/184459.606585:VERBOSE1:key_storage_util_linux.cc(54)] Password storage detected desktop environment: GNOME
[25344:1017/184459.606837:VERBOSE1:key_storage_linux.cc(128)] Selected backend for OSCrypt: GNOME_ANY

**Comment by rzhao271** [comment]
Did the password prompt dialog pop up behind the VS Code window?

**Comment by vregazzi** [comment]
No, it did not. The furthest I get is loading on "Select an account to sign in".
![image](https://github.com/microsoft/vscode/assets/118196285/be776b6a-d68a-40a7-a337-4e8a000513cf)

**Comment by TylerLeonhardt** [comment]
Do you have a keyring app installed like gnome-keyring, kwallet, etc? We're looking for something that could interact with the Secret Service of the OS: https://specifications.freedesktop.org/secret-service/latest/

**Comment by vregazzi** [comment]
I have gnome-keyring since it came preinstalled.

**Comment by TylerLeonhardt** [comment]
Two questions:
* Do you have libsecret installed?
* Can you confirm that the Login keyring is unlocked? (you can use a GUI like [seahorse](https://wiki.gnome.org/Apps/Seahorse) to check)

**Comment by vregazzi** [comment]
Libsecret is not installed. There is not a login keyring to be found in seahorse.

**Comment by vregazzi** [comment]
Should that have an effect on signing in?

**Comment by MrCodingB** [comment]
Had the same issue, but it was resolved after I rebooted the system. When I first had the issue it also helped to start code as root, but then the Firefox-popup for the GitHub login couldn't be opened.

**Comment by TylerLeonhardt** [comment]
@vregazzi just to rule out some factors, can you try using the basic text encryption option:

https://code.visualstudio.com/docs/editor/settings-sync#_not-recommended-configure-basic-text-encryption

If this works, then the issue has to be related to your systems setup of the keyring.

You can also try the other `gnome` options as documented here:

https://code.visualstudio.com/docs/editor/settings-sync#_recommended-configure-the-keyring-to-use-with-vs-code

Let me know how this goes

**Comment by qwertyaya** [comment]
I have the same issue, i have fixed it by running it as root with "--no-sandbox --user-data-dir" and specified my desktop directory just to see if it works and it does, hope that helps

**Comment by vregazzi** [comment]
> https://code.visualstudio.com/docs/editor/settings-sync#_not-recommended-configure-basic-text-encryption

Basic text encryption worked. Thank you so much for your help.

**Comment by TylerLeonhardt** [comment]
I really wish we could understand what was going wrong for you. It does seem like the OS is trying to show a dialog to unlock the keyring, but it's being hidden somehow. Basic text avoids using the OS keyring which is why this is "fixed".

**Comment by Tyriar** [comment]
I can reproduce this from a brand new Debian install, we definitely should be doing something better than crashing

**Comment by TylerLeonhardt** [comment]
* Run with the basic secret store: `code --password-store=basic`
* run this command `Developer: Open Logs Folder`
* run this command `Developer: Set Log Level`
* Click on the _double checkmark button_ ✔️✔️ on the same line as `Trace`
* quit VS Code
* start VS Code without basic secret store to trigger the hang again:
```
code --verbose --vmodule="*/components/os_crypt/*=1"
```
* Going back to the logs folder, give me what’s in the `main.log` and whatever is in the terminal

**Comment by Power-tile** [comment]
Facing the same issue on Ubuntu 22.04 and VSCode with version

```
$ code --version
1.104.0
f220831ea2d946c0dcb0f3eaa480eb435a2c1260
x64
```

However, basic password store does not solve the issue. Also, the **entire Ubuntu system crashes** because of this...

**Comment by Power-tile** [comment]
> * Run with the basic secret store: `code --password-store=basic`
> * run this command `Developer: Open Logs Folder`
> * run this command `Developer: Set Log Level`
> * Click on the _double checkmark button_ ✔️✔️ on the same line as `Trace`
> * quit VS Code
> * start VS Code without basic secret store to trigger the hang again:
> 
> ```
> code --verbose --vmodule="*/components/os_crypt/*=1"
> ```
> 
> * Going back to the logs folder, give me what’s in the `main.log` and whatever is in the terminal

Following these steps, I signed out and signed in to Settings Sync through Github. After the browser prompts to open link in vscode and I agree, the program stops responding. Then, clicking on the system time crashes the entire system. After rebooting, `main.log` contains:

[main.log](https://github.com/user-attachments/files/22308203/main.log)

There is no terminal due to the crash of the entire system. However, the crashed system shows the following three lines of errors infinitely with increasing ID in brackets:

```
[146.136898] systemd-journald[302]: Failed to rotate /var/log/journal/9251772f2ff846bbad0bd5eea842fab5/system.journal: Input/output error
[146.137400] systemd-journald[302]: Failed to rotate /var/log/journal/9251772f2ff846bbad0bd5eea842fab5/user-1000.journal: Input/output error
[146.137881] systemd-journald[302]: Faild to write entry (22 items, 670 bytes) despite vacuuming, ignoring: Input/output error
... and this log continues with increasing ID but same three lines repeatedly
```

**Comment by Power-tile** [comment]
Problem persists with Ubuntu 24.04.3 LTS and VSCode 1.104.0.
